contentcuration/contentcuration/viewsets/file.py

```diff
@@ -136,15 +136,14 @@ def upload_url(self, request):
             file_format = request.data["file_format"]
             preset = request.data["preset"]
         except KeyError:
-            raise HttpResponseBadRequest(
+            return HttpResponseBadRequest(
                 reason="Must specify: size, checksum, name, file_format, and preset"
             )
 
         try:
             request.user.check_space(float(size), checksum)
-
-        except PermissionDenied as e:
-            return HttpResponseBadRequest(reason=str(e), status=418)
+        except PermissionDenied:
+            return HttpResponseBadRequest(reason="Not enough space. Check your storage under Settings page.", status=400)
```

Comment:
I would suggest using a 412 code here specifically, as it seems to be the most appropriate:
https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/412

(the precondition being that the user's free storage is not big enough!)